tests/scripts_run/test_main.py

```diff
@@ -0,0 +1,774 @@
+        json_end = command.find("' --sweep_id")
+        config_json = json.loads(command[json_start:json_end])
+
+        assert config_json["wandb_project"] == "test-project"  # Should be overridden
+        assert config_json["C"] == 10
+        assert config_json["task_config"]["task_name"] == "tms"
+
+    @patch("spd.scripts.run.load_config")
+    @patch("spd.scripts.run.load_sweep_params")
+    def test_generate_commands_with_sweep(self, mock_load_sweep_params, mock_load_config):
+        """Test generate_commands function with sweep parameters."""
+        from spd.configs import Config
+
+        # Mock config loading
+        mock_config = Mock(spec=Config)
+        config_dict = get_valid_tms_config()
+        mock_config.model_dump.return_value = config_dict
+        mock_load_config.return_value = mock_config
+
+        # Mock sweep parameters
+        mock_load_sweep_params.return_value = {"C": {"values": [5, 10]}}
+
+        commands = generate_commands(
+            experiments_list=["tms_5-2-id"],
+            run_id="test_run_123",
+            sweep_params_file="sweep_params.yaml",
+            project="test-project",
+        )
+
+        assert len(commands) == 2  # Two C values
+
+        for i, command in enumerate(commands):
+            assert "python" in command
+            assert "tms_decomposition.py" in command
+            assert "--sweep_id test_run_123" in command
+            assert "--evals_id tms_5-2-id" in command
+            assert "--sweep_params_json" in command
+
+            # Extract and verify the config JSON
+            json_start = command.find("json:") + 5
+            json_end = command.find("' --sweep_id")
+            config_json = json.loads(command[json_start:json_end])
+
+            # Check that C was overridden
+            expected_c = [5, 10][i]
+            assert config_json["C"] == expected_c
+
+
+class TestLocalExecution:
+    """Test local execution flow."""
+
+    @patch("spd.scripts.run.create_wandb_report")
+    @patch("spd.scripts.run.create_workspace_view")
+    @patch("spd.scripts.run.ensure_project_exists")
+    @patch("spd.scripts.run.logger")
+    @patch("spd.scripts.run.subprocess.run")
+    def test_local_run_single_experiment(
+        self, mock_subprocess, mock_logger, mock_ensure_project, mock_workspace, mock_report
+    ):
+        """Test running a single experiment locally."""
+        # Setup mocks
+        mock_subprocess.return_value = Mock(returncode=0)
+        mock_workspace.return_value = "https://wandb.ai/test/workspace/tms_5-2-id"
+
+        # Call main with tms_5-2-id experiment in local mode
+        main(experiments="tms_5-2-id", local=True)
+
+        # Verify subprocess was called
+        assert mock_subprocess.call_count == 1
+
+        # Verify the command structure
+        args = mock_subprocess.call_args[0][0]
+        assert args[0] == "python"
+        assert "tms_decomposition.py" in args[1]
+
+        # Verify logger calls
```

Comment:
I'd remove all logger verification in this and any other test it's in. We don't want to have to change the tests when changing the logs.